WrongMethodTypeException: MethodHandle()KdbTree should be of type ()KdbTree

This exception occurs when spatial join is executed with ST_DISTANCE used in ON clause of inner join:

```
java.lang.BootstrapMethodError: CallSite bootstrap method initialization exception
	at java.base/java.lang.invoke.CallSite.makeSite(CallSite.java:336)
	at java.base/java.lang.invoke.MethodHandleNatives.linkCallSiteImpl(MethodHandleNatives.java:258)
	at java.base/java.lang.invoke.MethodHandleNatives.linkCallSite(MethodHandleNatives.java:248)
	at io.prestosql.$gen.PageProjectionWork_20210113_003224_42197.evaluate(Unknown Source)
	at io.prestosql.$gen.PageProjectionWork_20210113_003224_42197.process(Unknown Source)
	at io.prestosql.operator.project.DictionaryAwarePageProjection$DictionaryAwarePageProjectionWork.processInternal(DictionaryAwarePageProjection.java:185)
	at io.prestosql.operator.project.DictionaryAwarePageProjection$DictionaryAwarePageProjectionWork.process(DictionaryAwarePageProjection.java:121)
	at io.prestosql.operator.project.PageProcessor$ProjectSelectedPositions.processBatch(PageProcessor.java:333)
	at io.prestosql.operator.project.PageProcessor$ProjectSelectedPositions.process(PageProcessor.java:205)
	at io.prestosql.operator.WorkProcessorUtils$ProcessWorkProcessor.process(WorkProcessorUtils.java:372)
	at io.prestosql.operator.WorkProcessorUtils.lambda$flatten$6(WorkProcessorUtils.java:277)
	at io.prestosql.operator.WorkProcessorUtils$3.process(WorkProcessorUtils.java:319)
	at io.prestosql.operator.WorkProcessorUtils$ProcessWorkProcessor.process(WorkProcessorUtils.java:372)
	at io.prestosql.operator.WorkProcessorUtils$3.process(WorkProcessorUtils.java:306)
	at io.prestosql.operator.WorkProcessorUtils$ProcessWorkProcessor.process(WorkProcessorUtils.java:372)
	at io.prestosql.operator.WorkProcessorUtils.getNextState(WorkProcessorUtils.java:221)
	at io.prestosql.operator.WorkProcessorUtils.lambda$processStateMonitor$2(WorkProcessorUtils.java:200)
	at io.prestosql.operator.WorkProcessorUtils$ProcessWorkProcessor.process(WorkProcessorUtils.java:372)
	at io.prestosql.operator.WorkProcessorOperatorAdapter.getOutput(WorkProcessorOperatorAdapter.java:188)
	at io.prestosql.operator.Driver.processInternal(Driver.java:379)
	at io.prestosql.operator.Driver.lambda$processFor$8(Driver.java:283)
	at io.prestosql.operator.Driver.tryWithLock(Driver.java:675)
	at io.prestosql.operator.Driver.processFor(Driver.java:276)
	at io.prestosql.execution.SqlTaskExecution$DriverSplitRunner.processFor(SqlTaskExecution.java:1076)
	at io.prestosql.execution.executor.PrioritizedSplitRunner.process(PrioritizedSplitRunner.java:163)
	at io.prestosql.execution.executor.TaskExecutor$TaskRunner.run(TaskExecutor.java:484)
	at io.prestosql.$gen.Presto_346_5____20210112_131136_2.run(Unknown Source)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:834)
Caused by: java.lang.invoke.WrongMethodTypeException: MethodHandle()KdbTree should be of type ()KdbTree
	at java.base/java.lang.invoke.CallSite.wrongTargetType(CallSite.java:201)
	at java.base/java.lang.invoke.CallSite.makeSite(CallSite.java:325)
	... 29 more
```

Here's a simple repro:

```
CREATE SCHEMA hive.tmp;
USE hive.tmp;

CREATE TABLE spatial_partition_example AS
WITH data AS (
    SELECT * FROM (VALUES (41, 72), (41.1, 72.1), (41.2, 72.2), (41.3, 72.4)) as t (lat, lng)
)
SELECT spatial_partitioning(ST_Point(lat, lng), 10) as v
FROM data;

SET SESSION spatial_partitioning_table_name='spatial_partition_example';

WITH data AS (
    SELECT * FROM (VALUES (41, 72), (41.1, 72.1), (41.2, 72.2), (41.3, 72.4)) as t (lat, lng)
)
SELECT 
    a.lat,
    a.lng,
    ST_Distance(ST_Point(a.lat, a.lng), ST_Point(b.lat, b.lng)) distance
    FROM data a
    INNER JOIN data b ON
    ST_Distance(ST_Point(a.lat, a.lng), ST_Point(b.lat, b.lng)) < 0.2
```

--- a/plugin/trino-geospatial/src/main/java/io/trino/plugin/geospatial/KdbTreeType.java
+++ b/plugin/trino-geospatial/src/main/java/io/trino/plugin/geospatial/KdbTreeType.java
@@ -32,7 +32,10 @@ public final class KdbTreeType
 
     private KdbTreeType()
     {
-        super(new TypeSignature(NAME), KdbTree.class);
+        // The KDB tree type should be KdbTree but can not be since KdbTree is in
+        // both the plugin class loader and the system class loader.  This was done
+        // so the plan optimizer can process geo spatial joins.
+        super(new TypeSignature(NAME), Object.class);
     }
 
     @Override

